File: RegularExpressions/ruby/regularExpressions.rb
def ejemplo_1
    print "Ingrese una cadena de texto: "
    cadena = gets.chomp
    if cadena =~ /1[^2]/
        puts "Cadena VALIDA\n\n"
    else
        puts "NO VALIDO\n\n"
    end
end

def ejemplo_2
    print "Ingrese una cadena de texto: "
    cadena = gets.chomp
    if cadena !~ /\d$/
        puts "Cadena VALIDA\n\n"
    else
        puts "Termina con un dígito (NO VALIDO)\n\n"
    end
end

def mostrar_menu
    puts "MENU"
    puts "1. Ejemplo 1: Comprobar si el String cadena contiene un '1' que no está seguido por un '2'"
    puts "2. Ejemplo 2: Comprobar si el String cadena no acaba con un dígito"
    puts "3. Salir"
end

continuar = true
while continuar
    mostrar_menu
    print "Elija una opción: "
    opcion = gets.chomp
    case opcion
    when '1'
        ejemplo_1
    when '2'
        ejemplo_2
    when '3'
        puts "Saliendo del programa..."
        continuar = false
    else
        puts "Opción inválida, por favor elija una opción del menú\n\n"
    end
end

File: RegularExpressions/javascript/regularExpressions.js
const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function showMenu() {
    console.log("MENU")
    console.log("1. Validar Email")
    console.log("2. Comprobar tamaño de string")
    console.log("3. Reemplazar letras 'a' y 'c'")
    console.log("4. Salir")
    getMenuOption();
}
function getMenuOption() {
    rl.question('Elija una opcion: ', (opcion) => {
        switch (opcion) {
            case '1':
                // Ejemplo 1: Validar email
                const emailPattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
                rl.question('Ingresa un email: ', (email) => {
                    const match = emailPattern.test(email);
                    if (match) {
                        console.log("Email VALIDO\n\n");
                    } else {
                        console.log("Email NO VALIDO\n\n")
                    }
                    showMenu();
                })
                break;
            case '2': 
                //Comprobar si el String cadena está formado por un mínimo de 5 letras mayúsculas o minúsculas y un máximo de 10.
                const stringPattern = /^[a-zA-Z]{5,10}$/;
                rl.question('Escribe un string y se validará si tiene un valor minimo de 5 y un maximo de 10: ', (userText) => {
                    const match = stringPattern.test(userText);
                    if (match) {
                        console.log('La cadena CUMPLE con los requisitos\n\n')
                    } else {
                        console.log('La cadena NO CUMPLE con los requisitos\n\n')
                    }
                    showMenu();
                });
                
                break;
            case '3':
                // Ejemplo 2: Reemplazar las letras a y c en una cadena
                const pattern2 = /a?c|c?a/g;
                rl.question('Ingrese un texto y se reemplazaran las letras a y c: ', (text2) => {
                    newText = text2.replace(pattern2, 'X');
                    console.log('Texto modificado:', newText, "\n\n");
                    showMenu();
                })
                break;
            case '4':
                console.log("Saliendo del programa");
                break;
            default:
                console.log("Elija una opcion valida\n\n");
                showMenu();
                break;
        }
    });
}

showMenu();
getMenuOption();